doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -123,6 +123,11 @@ shows each existential with the hypotheses and conclusion for the associated goa
 
    Show Existentials.
 
+Users can control which goals are displayed in the context by :term:`focusing <focus>` and
+_shelving_ goals.  Focusing lets the user pick a single goal to work on.  Focusing operations
+can be nested.  See :ref:`focused_goals`.  Goals can be :term:`shelved` so they're not shown
+in the context until they are later :term:`unshelved`.  See :ref:`shelved_goals`.
```

Comment:
"so they're not shown in the context" seems to indicate that this is the purpose of shelving. The main reason for shelving goals is when they are expected to be solved by side effects (unification of existential variables happening while running tactics). Marginally, shelving can be used as a trick for other reasons.